src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/RazorCodeGenerationOptionsBuilder.cs

```diff
@@ -136,15 +136,6 @@ public bool UseEnhancedLinePragma
     /// </summary>
     public string? SuppressUniqueIds { get; set; }
 
-    /// <summary>
-    /// Determines whether RenderTreeBuilder.AddComponentParameter should not be used.
-    /// </summary>
-    public bool SuppressAddComponentParameter
-    {
-        get => _flags.IsFlagSet(RazorCodeGenerationOptionsFlags.SuppressAddComponentParameter);
-        set => _flags.UpdateFlag(RazorCodeGenerationOptionsFlags.SuppressAddComponentParameter, value);
-    }
```

Comment:
❗If this is removed and no longer set, `RazorCodeGenerationOptionsFlags.SuppressAddComponentParameter` and `RazorCodeGenerationOptions.SuppressAddComponentParameter` should also be removed. After all, this property is here to set a value on the underlying `RazorCodeGenerationOptions` that is being built. However, removing `RazorCodeGenerationOptions.SuppressAddComponentParameter` isn't possible.

`RazorCodeGenerationOptions.SuppressAddComponentParameter` is checked during code gen. If it's not set on the builder, that option will now report the wrong value during code gen. I suspect that the unit test failures are due to this no longer being set in the source generator.

Instead of removing this, a less invasive change would be to leave it here and then defer to `RazorConfiguration` when setting options in the source generator. I added a comment in `src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/SourceGenerators/RazorSourceGenerator.Helpers.cs` below that describes the change that I _think_ will make everything work.